builtin/providers/dns/data_dns_a_record.go

```diff
@@ -0,0 +1,77 @@
+package dns
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"net"
+	"sort"
+)
+
+func dataSourceDnsARecord() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceDnsARecordRead,
+		Schema: map[string]*schema.Schema{
+			"host": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			// Optionally sort A records in alphabetical order.
+			// This is helpful when a name uses round-robin DNS, which may
+			// sort records with multiple addresses in a non-deterministic order.
+			// This random sorting can cause flapping in terraform plans, where
+			// the changes in sort order cause dependent resources to update
+			// despite having no real change in the set of addresses.
+			"sort": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
```

Comment:
My inner self was telling me to let the developer have the ability to choose.
As you challenge it, I can't honestly think of a case where someone would want to see terraform plans change just because the records changed order.

I am going to remove sort, always sort the records.
If somebody wants different, they can always add it in later.